examples/neural_quantum_states/src/models/alt_transformer.py

```diff
@@ -0,0 +1,168 @@
+        self.input_order = np.stack([np.arange(self.num_sites-2,-1,-2), np.arange(self.num_sites-1,-1,-2)],1).reshape(-1) # [4,5,2,3,0,1]
+        self.input_order = torch.Tensor(self.input_order).int().to(self.device)
+        self.shell_order = torch.arange(self.num_sites//2-1, -1, -1) # [2,1,0]
+        
+        transformer_layer = nn.TransformerEncoderLayer(embedding_dim, nhead, dim_feedforward=dim_feedforward, dropout=0.0, batch_first=True)
+        self.transformer = nn.TransformerEncoder(transformer_layer, num_layers)
+        self.fc = nn.Linear(embedding_dim, 4)
+        self.tok_emb = nn.Embedding(5, embedding_dim)
+        self.pos_emb = nn.Embedding(len(self.shell_order), embedding_dim)
+        self.apply(self._init_weights)
+        self.softmax = nn.Softmax(dim=-1)
+        self.log_softmax = nn.LogSoftmax(dim=-1)
+
+        self.net_phase = nn.Linear(in_features=embedding_dim*len(self.shell_order), out_features=4, bias=True)
+
+        self.mask = torch.zeros((len(self.shell_order), len(self.shell_order))).to(self.device)
+        for i in range(len(self.mask)):
+            for j in range(len(self.mask)):
+                if i < j:
+                    self.mask[i][j] = float('-inf') 
+        
+    def _init_weights(self, module: nn.Module):
+        if isinstance(module, nn.Linear):
+            torch.nn.init.normal_(module.weight, mean=0.0, std=0.02)
+            if module.bias is not None:
+                torch.nn.init.zeros_(module.bias)
+        elif isinstance(module, nn.Embedding):
+            torch.nn.init.normal_(module.weight, mean=0.0, std=0.02)
+        elif isinstance(module, nn.LayerNorm):
+            torch.nn.init.zeros_(module.bias)
+            torch.nn.init.ones_(module.weight)
+
+    def forward(self, x: torch.Tensor, sample_shell: int = 0) -> torch.Tensor:
+        '''
+        Forward function for Transformer ansatz (used for both sampling and training)
+        Args:
+            x: qubit spin configuration
+            sample_shell: shell index provided during sampling to avoid extraneous forward passes
+        Returns:
+            prob_cond/log_psi: either conditional probabilities of logarithms of statevector entries, depending on if sampling
+        '''
+        # x: [bs, num_sites]
+        shells = self.state2shell(x)[:, self.shell_order]
+        input = 4*torch.ones(shells.shape, dtype=torch.int64).to(self.device) 
+        input[:,1:] = shells[:,:-1]
+        pos = self.shell_order.to(self.device)
+
+        input = self.tok_emb(input) + self.pos_emb(pos)
+        # new x is of shape (batch_size, sequence_length, d_model)
+
+        if self.mask.device != self.device:
+            self.mask = self.mask.to(self.device)
+        output = self.transformer(input[:,:(len(self.shell_order) - sample_shell + 1)], mask=self.mask[:(len(self.shell_order) - sample_shell + 1),:(len(self.shell_order) - sample_shell + 1)], is_causal=True)
+        if not self.sampling:
```

Comment:
I would recommend adding some spaces in-between `if` blocks for better readability.